gms/gossiper.cc

```diff
@@ -2120,15 +2120,20 @@ future<> gossiper::add_saved_endpoint(inet_address ep) {
     if (!tokens.empty()) {
         std::unordered_set<dht::token> tokens_set(tokens.begin(), tokens.end());
         ep_state.add_application_state(gms::application_state::TOKENS, versioned_value::tokens(tokens_set));
+        ep_state.add_application_state(gms::application_state::STATUS, gms::versioned_value::shutdown(true));
```

Comment:
If it is removed, how come it owns tokens in system.peers?

Reply:
It could happen like this:
1. shutdown nodes A, B
2. on node C perform `--removenode A --ignore-dead-nodes B`
3. node A is removed, but on node B it still owns tokens.
4. Restart node B. Node A gets resurrected.

The question is, if some node was used inside `--ignore-dead-nodes`, maybe we should assume it is also permanently dead.

Reply:
> The question is, if some node was used inside --ignore-dead-nodes, maybe we should assume it is also permanently dead.

dead = permanently down

It's only safe to ignore a node if it's permanently down.

Reply:
We can also extend removenode to remove multiple nodes at once, but I'm not sure we want to go there.

Reply:
Another alternative is to provide an api to declare nodes as DEAD before removenode or replace.

Reply:
This can be record persistently and be used to fence them off permanently.